jena-arq/src/main/java/org/apache/jena/sparql/engine/join/AbstractIterHashJoin.java

```diff
@@ -234,13 +234,11 @@ protected void closeSubIterator() {
         if ( JoinLib.JOIN_EXPLAIN ) {
             String x = String.format(
                          "HashJoin: LHS=%d RHS=%d Results=%d Table=%s",
-                         s_countProbe, s_countScan, s_countResults,
-                         hashTable.toString()) ;
+                         s_countProbe, s_countScan, s_countResults, hashTable) ;
             System.out.println(x) ;
         }
         // In case it's a peek iterator.
         iterStream.close() ;
-        hashTable.clear();
```

Comment:
The hashTable does not manage any resources - so there is no need to do the work to clear it. It can just be gc'd with the iterator.

Reply:
The heap subsystem is more than just GC'ing. Clearing objects can release memory earlier (minor GC) or (unlikely here) th eheap sees short lived objects and does effectively a malloc free.

It is hard to tell if it makes a real difference, but it isn't complicated to do here.

And `hashTable = null;` It would also be an internal consistency check to make sure it isn't mistakenly reused.

This is single thread code so a simple solution 

```java
        if ( hashTable != null ) {
            hashTable.clear();
            hashTable = null;
        }
```
either harmless or may be an advantage.